streampark-common/src/main/scala/org/apache/streampark/common/conf/SparkVersion.scala

```diff
@@ -36,7 +38,14 @@ class SparkVersion(val sparkHome: String) extends Serializable with Logger {
 
   private[this] lazy val SPARK_SCALA_VERSION_PATTERN = Pattern.compile("^spark-core_(.*)-[0-9].*.jar$")
 
-  lazy val scalaVersion: String = SPARK_SCALA_VERSION_PATTERN.matcher(sparkCoreJar.getName).group(1)
+  lazy val scalaVersion: String = {
```

Comment:
It would be better to throw an exception if the Scala version cannot be extracted.
```
  lazy val scalaVersion: String = {
    val matcher = SPARK_SCALA_VERSION_PATTERN.matcher(sparkCoreJar.getName)
    if (matcher.find()) {
      matcher.group(1)
    } else {
      throw new IllegalArgumentException(s"[StreamPark] can not found scala version in $sparkCoreJar")
    }
  }
```

Reply:
done